sibyl/learn/tracer/pin.py

```diff
@@ -0,0 +1,99 @@
+'''
+This module gives a tracer that uses pin (3.0-76991-gcc-linux) to run the program
+'''
+
+import tempfile
+import os
+
+import sibyl
+from sibyl.learn.tracer.tracer import Tracer
+from sibyl.learn.trace import Trace, Snapshot
+
+
+class TracerPin(Tracer):
+
+    '''Tracer that uses pin'''
+
+    def do_trace(self):
+        '''Run the pintool and construct the trace from the pintool output file'''
+
+        tmpName = self.__run_pin_cmd()
+        return self.__parse_pin_output_file(open(tmpName))
+
+    def __run_pin_cmd(self):
+        '''Run the pintool'''
+
+        tmp = tempfile.NamedTemporaryFile(suffix=".trace", delete=False)
+        tmpName = tmp.name
+        tmp.close()
+
+        cmd = ["pin", "-t", os.path.dirname(sibyl.__path__[0]) + "/ext/pintool/pintool.so", "-a", hex(self.address), "-o", tmpName, "--", "./" + self.program]
```

Comment:
`hex(self.address)` can be replaced with a `"0x%x" % self.address`, to avoid the long notation of Python ('0x...L')